heir_py/openfhe_config.py

```diff
@@ -0,0 +1,89 @@
+"""Configuration of OpenFHE backend."""
+
+import os
+import subprocess
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class OpenFHEConfig:
+    # The include paths for OpenFHE headers
+    include_dirs: list[str]
+
+    # The directory containing libOPENFHEbinfhe.so, etc.
+    lib_dir: str
+
+    # The libraries to link against (without lib prefix or .so suffix)
+    link_libs: list[str]
+
+    # The type of include paths to use during codegen. Options are:
+    # - "install-relative": use paths relative to the installed OpenFHE
+    # - "source-relative": relative to the openfhe development repository.
+    include_type: str = "install-relative"
+
+
+DEFAULT_INSTALLED_OPENFHE_CONFIG = OpenFHEConfig(
```

Comment:
just feedback for another time: i think it might be worth adding a warning incase these hardcoded dirs don't exist so that the user gets early feedback before clang tries to compile with these flags